2025.4.0 generation failing

Hard to say from the GitHub Action logs, but this seems to be the error that occurs during stubgen:

> AssertionError: Internal error: unresolved placeholder type None

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -53,7 +53,7 @@ prerelease = "allow"
 
 [tool.uv.sources]
 # TODO switch back to released versions with 1.16.0
-mypy = { git = "https://github.com/python/mypy", rev = "3ced11a43e75dd97554d5c7af78da296ee0d04ec" }
+mypy = { git = "https://github.com/python/mypy", rev = "e3680ce169e31a5c7d268e022eb1501e93f6fdba" }
 
 [tool.pylint.format]
 expected-line-ending-format = "LF"

--- a/update_stubs.py
+++ b/update_stubs.py
@@ -325,7 +325,7 @@ def get_old_typed_paths(homeassistant_root: Path) -> list[Path]:
 def generate_stubs(typed_paths: list[Path], repo_root: Path) -> None:
     """Use stubgen to generate typing stubs for all typed paths."""
     LOGGER.info("Generating stubs...")
-    with tempfile.NamedTemporaryFile("w", delete_on_close=False) as f:
+    with tempfile.NamedTemporaryFile("w", delete_on_close=False, delete=False) as f:
         f.write("\n".join(map(str, typed_paths)))
         f.close()
         command_args: list[str] = [

